src/ApplicationLogs/Store/LogStorageStore.cs

```diff
@@ -0,0 +1,373 @@
+
+        #endregion
+
+        #region Put
+
+        public void PutBlockState(UInt256 hash, TriggerType trigger, BlockLogState state)
+        {
+            var key = new KeyBuilder(Prefix_Id, Prefix_Block)
+                .Add(hash)
+                .Add((byte)trigger)
+                .ToArray();
+            _snapshot.Put(key, state.ToArray());
+        }
+
+        public Guid PutNotifyState(NotifyLogState state)
+        {
+            var id = Guid.NewGuid();
+            var key = new KeyBuilder(Prefix_Id, Prefix_Notify)
+                .Add(id.ToByteArray())
+                .ToArray();
+            _snapshot.Put(key, state.ToArray());
+            return id;
+        }
+
+        public void PutContractState(UInt160 scriptHash, ulong timestamp, uint iterIndex, ContractLogState state)
+        {
+            var key = new KeyBuilder(Prefix_Id, Prefix_Contract)
+                .Add(scriptHash)
+                .AddBigEndian(timestamp)
+                .AddBigEndian(iterIndex)
+                .ToArray();
+            _snapshot.Put(key, state.ToArray());
+        }
+
+        public Guid PutExecutionState(ExecutionLogState state)
+        {
+            var id = Guid.NewGuid();
+            var key = new KeyBuilder(Prefix_Id, Prefix_Execution)
+                .Add(id.ToByteArray())
+                .ToArray();
+            _snapshot.Put(key, state.ToArray());
+            return id;
+        }
+
+        public void PutExecutionBlockState(UInt256 blockHash, TriggerType trigger, Guid executionStateId)
+        {
+            var key = new KeyBuilder(Prefix_Id, Prefix_Execution_Block)
+                .Add(blockHash)
+                .Add((byte)trigger)
+                .ToArray();
+            _snapshot.Put(key, executionStateId.ToByteArray());
+        }
+
+        public void PutExecutionTransactionState(UInt256 txHash, Guid executionStateId)
+        {
+            var key = new KeyBuilder(Prefix_Id, Prefix_Execution_Transaction)
+                .Add(txHash)
+                .ToArray();
+            _snapshot.Put(key, executionStateId.ToByteArray());
+        }
+
+        public void PutTransactionState(UInt256 hash, TransactionLogState state)
+        {
+            var key = new KeyBuilder(Prefix_Id, Prefix_Transaction)
+                .Add(hash)
+                .ToArray();
+            _snapshot.Put(key, state.ToArray());
+        }
+
+        public Guid PutStackItemState(StackItem stackItem)
+        {
+            var id = Guid.NewGuid();
+            var key = new KeyBuilder(Prefix_Id, Prefix_StackItem)
+                .Add(id.ToByteArray())
+                .ToArray();
+            try
+            {
+                _snapshot.Put(key, BinarySerializer.Serialize(stackItem, (uint)Settings.Default.MaxStackSize));
+            }
+            catch (NotSupportedException)
+            {
+                _snapshot.Put(key, BinarySerializer.Serialize(StackItem.Null, (uint)Settings.Default.MaxStackSize));
+            }
```

Comment:
`Now` changed to `public static byte[] Serialize(StackItem item, ExecutionEngineLimits limits)` in https://github.com/neo-project/neo/pull/2948